app/controllers/search_controller.rb

```diff
@@ -4,40 +4,32 @@ class SearchController < SessionsController
   require 'will_paginate/array'
 
   def explore
-    @repositories = Repository.order([sort_order].to_h).page params[:page]
+    @repositories = Repository.paginate(:per_page=>12,:page=>params[:page]).order([sort_order].to_h).page params[:page]
     @photos = photo_hash
   end
 
   def search
   	sort_arr = sort_order
-  	repositories_by_attributes = Repository.where("lower(title) LIKE ?
+  	@repositories = Repository.paginate(:per_page=>12,:page=>params[:page]).where("lower(title) LIKE ?
                                                 OR lower(description) LIKE ?
                                                 OR lower(user_username) LIKE ?
                                                 OR lower(category) LIKE ?",
                                   "%#{params[:q].downcase}%",
                                   "%#{params[:q].downcase}%",
                                   "%#{params[:q].downcase}%",
-                                  "%#{params[:q].downcase}%")
-    repositories_by_categories = []
-    Category.where('lower(name) LIKE ?', params[:q].downcase).each do |cat|
-      repositories_by_categories << cat.repository
-	  end
-    @repositories = repositories_by_attributes + repositories_by_categories
-    @repositories = @repositories.uniq
-    @repositories.paginate(:per_page=>12,:page=>params[:page]) do
-      order_by sort_arr.first, sort_arr.last
-    end
+                                  "%#{params[:q].downcase}%").uniq
```

Comment:
This should be `distinct` which will make the db do the uniqueness stuff for you